src/request/components/request-detail-panel-execution.jsx

```diff
@@ -336,6 +338,60 @@ var ParamaterList = React.createClass({
     }
 });
 
+var MiddlewareList = React.createClass({
+    render: function() {
+        var middlewareStartMessages = this.props.middlewareStartMessages;
+        var middlewareEndMessages = this.props.middlewareEndMessages;
+        var beforeExecuteCommandMessages = this.props.beforeExecuteCommandMessages;
+        var afterExecuteCommandMessages = this.props.afterExecuteCommandMessages;
+        var mongoDBMessages = this.props.mongoDBMessages;
+        
+        var content = null;
+        
+        if (middlewareStartMessages
+            && middlewareEndMessages
+            && middlewareStartMessages.length >= 0
+            && middlewareEndMessages.length >= 0) {
+        
+            middlewareStartMessages = middlewareStartMessages.sort(function (a,b) { return a.ordinal - b.ordinal; });
+            middlewareEndMessages = _.indexBy(middlewareEndMessages, 'payload.correlationId');
```

Comment:
It is a base assumption across the platform that order "ordinal must also correspond to the order in which the events occurred". We should probably try documenting that more. In the .net implementation the `order` is determined in the request to help with this problem. I'm happy to open the conversation around this again, but we already know we have concrete cases where a concept like `correlationId` in the message payload won't work. Beyond making sure its documented for platform authors, we know we can get `order` right. Does that make sense/sound reasonable or I might be missing something?